### ISSUE TITLE
Support AppTrackingTransparency prompt and permission

### ISSUE THREAD

**Issue by sohail-dragon** [zoontek/react-native-permissions#500]
# Feature request

With iOS 14 launching soon, it'll be useful to support the AppTrackingTransparency framework to display the prompt for the user to  allow the app to track them (and make the IDFA available).

## Why it is needed

To support add tracking in iOS 14 and above

## Possible implementation

Use the AppTrackingTransparency framework along with `ATTrackingManager requestTrackingAuthorizationWithCompletionHandler` and `ATTrackingManager trackingAuthorizationStatus`.

### Code sample

PR incoming

**Comment by zoontek** [comment]
Released: https://github.com/react-native-community/react-native-permissions/releases/tag/2.2.0

**Comment by rochapablo** [comment]
I'm getting `Module 'AppTrackingTransparency' not found`

**Xcode Error**

```
node_modules/react-native-permissions/ios/AppTrackingTransparency/RNPermissionHandlerAppTrackingTransparency.m:3:9: Module 'AppTrackingTransparency' not found
```

**Podfile**

```
permissions_path = '../node_modules/react-native-permissions/ios'
pod 'Permission-AppTrackingTransparency', :path => "#{permissions_path}/AppTrackingTransparency.podspec"
```

**Info.plist**

```
<key>NSUserTrackingUsageDescription</key>
<string>This identifier will be used to deliver personalized ads to you.</string>
```

Then I ran `pod install`. Am I forgetting something?

**Comment by zoontek** [comment]
@rochapablo You need the new Xcode with iOS 14 support

**Comment by sohail-dragon** [comment]
Looks like we have a stay of execution on this change:
**tl;dr**
> To give developers time to make necessary changes, apps will be required to obtain permission to track users starting early next year. More information, including an update to the App Store Review Guidelines, will follow this fall.

Full article [here](https://developer.apple.com/news/?id=hx9s63c5)

**Comment by rochapablo** [comment]
@zoontek, thanks, the error is gone. But shouldn't I be getting the notification about AppTrackingTransparency?

**Comment by rochapablo** [comment]
Wait! Do I have to code the request for permission? Because I did and now it's working :neutral_face:

```
request(PERMISSIONS.IOS.APP_TRACKING_TRANSPARENCY)
```
